Sources/XcodeProjMapper/Mappers/Graph/XcodeGraphMapper.swift

```diff
@@ -0,0 +1,282 @@
+import FileSystem
+import Foundation
+import Path
+import PathKit
+import XcodeGraph
+import XcodeProj
+
+/// A protocol defining how to map a given file path to a `Graph`.
+public protocol XcodeGraphMapping {
+    /// Builds a `Graph` from the specified path.
+    /// - Parameter pathString: The absolute path to a `.xcworkspace`, `.xcodeproj`, or directory containing them.
+    /// - Returns: A `Graph` representing the projects, targets, and dependencies found at `pathString`.
+    /// - Throws: If the path doesn't exist or no projects are found.
+    func map(at pathString: AbsolutePath) async throws -> Graph
+}
+
+/// An error type for `XcodeGraphMapper` when the path is invalid or no projects are found.
+public enum XcodeGraphMapperError: LocalizedError {
+    case pathNotFound(String)
+    case noProjectsFound(String)
+
+    public var errorDescription: String? {
+        switch self {
+        case let .pathNotFound(path):
+            return "The specified path does not exist: \(path)"
+        case let .noProjectsFound(path):
+            return "No `.xcworkspace` or `.xcodeproj` was found at: \(path)"
+        }
+    }
+}
+
+/// Specifies whether we’re mapping a single `.xcodeproj` or an `.xcworkspace`.
+enum XcodeMapperGraphType {
+    case workspace(XCWorkspace)
+    case project(XcodeProj)
+}
+
+/// A unified entry point that locates `.xcworkspace` or `.xcodeproj` files—even within directories—and
+/// constructs a comprehensive `Graph` of projects, targets, and dependencies.
+///
+/// Specifically, this mapper:
+/// 1. Detects whether the input path is a single project, a workspace, or a directory.
+/// 2. Enumerates all discovered targets and dependencies to assemble the final `Graph`.
+///
+/// This replaces old parsers/providers with a single approach. For example:
+/// ```swift
+/// let mapper: XcodeGraphMapping = XcodeGraphMapper()
+/// let graph = try await mapper.map(at: "/path/to/MyApp")
+/// ```
+public struct XcodeGraphMapper: XcodeGraphMapping {
+    private let fileSystem: FileSysteming
+
+    // MARK: - Initialization
+
+    public init(fileSystem: FileSysteming = FileSystem()) {
+        self.fileSystem = fileSystem
+    }
+
+    // MARK: - Public API
+
+    public func map(at path: AbsolutePath) async throws -> Graph {
```

Comment:
I'd put this function also in the `XcodeGraphMapping` protocol.

Reply:
It is in the protocol. I updated the internal parameter name of the protocol to match. Must of missed it when updating and compiler doesn't care if internal ones match